### ISSUE TITLE
Error creating dataloaders: object of type 'generator' has no len()

### ISSUE THREAD

**Issue by Eupham** [Eupham/KernelDev#16]
Cloning into 'KernelDev'...
remote: Enumerating objects: 1143, done.
remote: Counting objects: 100% (13/13), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 1143 (delta 1), reused 6 (delta 1), pack-reused 1130 (from 2)
Receiving objects: 100% (1143/1143), 1.00 MiB | 8.40 MiB/s, done.
Resolving deltas: 100% (697/697), done.
Running in single process mode (nproc_per_node = 1).
Configuration loaded from: KernelDev/config.yaml
=== GPU Information ===
Device: NVIDIA H100 80GB HBM3
Compute Capability: (9, 0)
Total Memory: 79.2 GB
Current Memory Usage: 0.0 GB
Current Memory Cached: 0.0 GB
✓ H100-optimized flash attention kernels will be used

=== GPT Model Training with Flash Attention ===
Precision: bf16 (bfloat16)
Mixed Precision Training: Enabled (bf16)
Setting up configuration...

=== Initializing Model ===

=== Setting up Precision ===
Setting up mixed precision training (bf16)...
✓ Model prepared for bf16 mixed precision training
✓ Gradient scaler initialized for mixed precision
Total parameters: 810,027,008
Trainable parameters: 810,027,008
Parameter dtype: torch.float32

=== Memory Estimation ===
Estimated memory usage (bf16):
  Model parameters: 1.1GB
  Gradients: 2.3GB
  Optimizer states: 4.5GB
  Total model memory: 7.9GB
  Activation memory per sample: 93.8MB
  Available for activations: 69.1GB
  Recommended batch size: 737
Using estimated batch_size: 16
Device: cuda
Model config: {'vocab_size': 256, 'dim': 2048, 'n_layers': 12, 'n_heads': 16, 'max_seq_len': 2048, 'mlp_ratio': 4, 'causal': True}
Data config: {'dataset_name': 'allenai/c4', 'dataset_config': 'en', 'seq_len': 1024, 'max_samples': 1000000, 'max_eval_tokens': 50000, 'streaming_train': False, 'num_workers': 0}
Training config: batch_size=16, epochs=1

=== Loading and Processing Data ===
Using UTF-8 byte tokenization with vocabulary size: 256
Max evaluation tokens per split: 50000
Will attempt to load up to 1000000 samples from the dataset.
Loading dataset: allenai/c4/en
Streaming for 'train' split: False
Attempting to load 'train' split...
README.md: 41.1kB [00:00, 45.6MB/s]
Resolving data files: 100%|████████████████| 1024/1024 [00:04<00:00, 238.02it/s]
Resolving data files: 100%|██████████████| 1024/1024 [00:00<00:00, 34888.93it/s]
Failed to load 'train' split for allenai/c4: object of type 'generator' has no len()
Attempting to load 'validation' split (non-streaming, limited)...
Resolving data files: 100%|██████████████| 1024/1024 [00:00<00:00, 25768.82it/s]
Resolving data files: 100%|█████████████| 1024/1024 [00:00<00:00, 468218.39it/s]
Error creating dataloaders: object of type 'generator' has no len()

**Comment by Eupham** [comment]
=== Loading and Processing Data ===
Using UTF-8 byte tokenization with vocabulary size: 256
Max evaluation tokens per split: 50000
Will attempt to load up to 1000000 samples from the dataset.
Loading dataset: allenai/c4/en
Streaming for 'train' split: False
Attempting to load 'train' split...
Resolving data files: 100%|████████████████| 1024/1024 [00:05<00:00, 187.00it/s]
Resolving data files: 100%|██████████████| 1024/1024 [00:00<00:00, 93085.55it/s]
Failed to load 'train' split for allenai/c4: object of type 'generator' has no len()
Attempting to load 'validation' split (non-streaming, limited)...
Resolving data files: 100%|██████████████| 1024/1024 [00:00<00:00, 23289.69it/s]
Resolving data files: 100%|██████████████| 1024/1024 [00:00<00:00, 20333.81it/s]
Error creating dataloaders: object of type 'generator' has no len()
